config/zookeeper_persister.go

```diff
@@ -208,11 +208,31 @@ func (z *ZkConfigPersister) ReadPersistedConfig() (*pb.ServiceConfig, error) {
 }
 
 func (z *ZkConfigPersister) currentConfigEventListener() (<-chan zk.Event, error) {
+	if z.initialized {
+		logging.Printf("Re-establishing zookeeper watch on %v", z.path)
+	} else {
+		logging.Printf("Establishing zookeeper watch on %v", z.path)
+	}
+
+	// Ignoring the response to getting the contents of the watch. We don't care which node triggered the watch, since
+	// we're working out the "most recent" config below by iterating over all available configurations and sorting by
+	// the configuration's Version field. All we care about here is the channel watching the ZK path, to be notified of
+	// future changes.
+	// Placing this before the z.conn.Children() ensures we won't miss any new versions that are created during the
+	// search
+	_, _, ch, err := z.conn.GetW(z.path)
+
+	if err != nil {
+		logging.Printf("Received error from zookeeper when fetching %s: %+v", z.path, err)
+		return nil, err
+	}
+
 	if z.initialized {
```

Comment:
Will the value of `z.initialized` ever have changed from when it is checked above?

Reply:
I don't believe so